ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlDefinedIdentifiersVerifier.kt

```diff
@@ -118,4 +130,48 @@ internal class SamlDefinedIdentifiersVerifier(val node: Node) {
             }
         }
     }
+
+    /** 8.3.7 Persistent Identifier */
+    private fun verifyPersistentIdentifiers() {
+        node.recursiveChildren()
+            .filter { it.namespaceURI != null }
+            .filter { it.namespaceURI == PERSISTENT_ID }
+            .forEach {
+                if (it.textContent != null && it.textContent.length > ID_VALUE_LENGTH_LIMIT)
+                    throw SAMLComplianceException.create(SAMLCore_8_3_7_a,
+                        message = "The length of the Persistent ID's value was greater than " +
+                            "$ID_VALUE_LENGTH_LIMIT characters.",
+                        node = it)
+
+                if (it.children("NameQualifier")
+                        .any { it.textContent != idpMetadata.entityId })
+                    throw SAMLComplianceException.create(SAMLCore_8_3_7_b,
+                        SAMLCore_8_3_7_c,
+                        message = "The Persistent ID's NameQualifier isn't ${idpMetadata.entityId}",
+                        node = it)
+
+                if (it.children("SPNameQualifier")
```

Comment:
❗️ Aren't `NameQualifier` and `SPNameQualifier` attributes rather than child elements?
✏️ If the above is true: When fixing that, you could put the actual attributeText in the exception message instead of just putting the expected value.

Reply:
Correct. Fixed